ci-operator/step-registry/ipi/conf/azure/confidential/ipi-conf-azure-confidential-commands.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH="/tmp/install-config-security.yaml.patch"
+
+# OCPBUGS-36670: encryptionAtHost and settings.securityType are not related.
+# feature setting of image definition should only depend on settings.securityType
+if [[ "${ENABLE_CONFIDENTIAL_DEFAULT_MACHINE}" == "true" ]]; then
+    cat >> "${PATCH}" << EOF
+platform:
+  azure:
+    defaultMachinePlatform:
+      encryptionAtHost: false
+      settings:
+        securityType: ${AZURE_SECURITY_TYPE}
+        ${AZURE_SECURITY_TYPE,}:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+EOF
+    if [[ -n "${OS_DISK_SECURITY_ENCRYPTION_TYPE}" ]]; then
+        cat >> "${PATCH}" << EOF
+      osDisk:
+        securityProfile:
+          securityEncryptionType: ${OS_DISK_SECURITY_ENCRYPTION_TYPE}
+EOF
+    fi
+fi
+
+if [[ "${ENABLE_CONFIDENTIAL_CONTROL_PLANE}" == "true" ]]; then
+    cat >> "${PATCH}" << EOF
+controlPlane:
+  platform:
+    azure:
+      encryptionAtHost: false
+      settings:
+        securityType: ${AZURE_SECURITY_TYPE}
+        ${AZURE_SECURITY_TYPE,}:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+EOF
+    if [[ -n "${OS_DISK_SECURITY_ENCRYPTION_TYPE}" ]]; then
+        cat >> "${PATCH}" << EOF
+      osDisk:
+        securityProfile:
+          securityEncryptionType: ${OS_DISK_SECURITY_ENCRYPTION_TYPE}
+EOF
+    fi
+fi
+
+if [[ "${ENABLE_CONFIDENTIAL_COMPUTE}" == "true" ]]; then
+    cat >> "${PATCH}" << EOF
+compute:
+- platform:
+    azure:
+      encryptionAtHost: true
+      settings:
+        securityType: ${AZURE_SECURITY_TYPE}
+        ${AZURE_SECURITY_TYPE,}:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+EOF
+    if [[ -n "${OS_DISK_SECURITY_ENCRYPTION_TYPE}" ]]; then
```

Comment:
Can we move the block as a separate block when `OS_DISK_SECURITY_ENCRYPTION_TYPE` not empty.

Reply:
hmm, If moved to a separate block, I need to know where to insert osDisk setting, under defaultMachinePlatform or controlplane or compute?

Reply:
good point, you are right, ignore my comment.